src/libraries/System.Private.Xml/src/System/Xml/BinaryXml/XmlBinaryReader.cs

```diff
@@ -2349,28 +2344,21 @@ private int ScanText(out int start)
         private string GetString(int pos, int cch)
         {
             Debug.Assert(pos >= 0 && cch >= 0);
-            if (checked(pos + (cch * 2)) > _end)
+            if (checked(pos + (cch * sizeof(char))) > _end)
                 throw new XmlException(SR.Xml_UnexpectedEOF1, (string[]?)null);
             if (cch == 0)
                 return string.Empty;
-            // GetStringUnaligned is _significantly_ faster than unicode.GetString()
-            // but since IA64 doesn't support unaligned reads, we can't do it if
-            // the address is not aligned properly.  Since the byte[] will be aligned,
-            // we can detect address alignment my just looking at the offset
-            if ((pos & 1) == 0)
-                return GetStringAligned(_data, pos, cch);
-            else
-                return _unicode.GetString(_data, pos, checked(cch * 2));
-        }
 
-        private unsafe string GetStringAligned(byte[] data, int offset, int cch)
-        {
-            Debug.Assert((offset & 1) == 0);
-            fixed (byte* pb = data)
+            return string.Create(cch, (_data, pos), static (dstChars, state) =>
             {
-                char* p = (char*)(pb + offset);
-                return new string(p, 0, cch);
-            }
+                // bitblt source bytes directly into the destination char span
+                // n.b. source buffer assumed to be well-formed UTF-16 machine endian
+
+                int cch = dstChars.Length;
+                ReadOnlySpan<byte> srcBytes = state._data.AsSpan(state.pos, checked(cch * sizeof(char)));
+                Span<byte> dstBytes = MemoryMarshal.AsBytes(dstChars);
```

Comment:
Technically `MemoryMarshal` is an unsafe-equivalent API. But at least we got rid of the pointer manipulation in this method. :)